sql/planner/distinct.go

```diff
@@ -0,0 +1,43 @@
+package planner
+
+import (
+	"github.com/genjidb/genji/database"
+	"github.com/genjidb/genji/document"
+	"github.com/genjidb/genji/sql/query/expr"
+)
+
+type dedupNode struct {
+	node
+
+	tableName string
+	indexes   map[string]database.Index
+}
+
+func NewDistinctNode(n Node, tableName string) Node {
```

Comment:
We should rename this, Dedup is the name of an operation that could be used in several cases, DISTINCT is the name of a specific SQL feature.

```suggestion
func NewDedupNode(n Node, tableName string) Node {
```